src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -0,0 +1,142 @@
+
+    * :http-client-ssl-protocols - A list of legal SSL protocols that may be used
+        when https client requests are made.
+
+    * :http-client-cipher-suites - A list of legal SSL cipher suites that may
+        be used when https client requests are made."
+  {:ruby-load-path              (schema/both [schema/Str] (schema/pred vector?))
+   :gem-home                    schema/Str
+   :master-conf-dir             (schema/maybe schema/Str)
+   :master-var-dir              (schema/maybe schema/Str)
+   :http-client-ssl-protocols   [schema/Str]
+   :http-client-cipher-suites   [schema/Str]
+   :max-active-instances        schema/Int
+   :max-requests-per-instance   schema/Int})
+
+(def JRubyPoolAgent
+  "An agent configured for use in managing JRuby pools"
+  (schema/both Agent
+               (schema/pred
+                 (fn [a]
+                   (let [state @a]
+                     (and
+                       (map? state)
+                       (ifn? (:shutdown-on-error state))))))))
+
+(def PoolState
+  "A map that describes all attributes of a particular JRubyPuppet pool."
+  {:pool         pool-queue-type
+   :size schema/Int})
+
+(def PoolStateContainer
+  "An atom containing the current state of all of the JRubyPuppet pool."
+  (schema/pred #(and (instance? Atom %)
+                     (nil? (schema/check PoolState @%)))
+               'PoolStateContainer))
+
+(def PoolContext
+  "The data structure that stores all JRubyPuppet pools and the original configuration."
+  {:config                JRubyPuppetConfig
+   :profiler              (schema/maybe PuppetProfiler)
+   :pool-agent            JRubyPoolAgent
+   :flush-instance-agent  JRubyPoolAgent
+   :pool-state            PoolStateContainer})
+
+(def JRubyInstanceState
+  "State metadata for an individual JRubyPuppet instance"
+  {:request-count schema/Int})
+
+(def JRubyInstanceStateContainer
+  "An atom containing the current state of a given JRubyPuppet instance."
+  (schema/pred #(and (instance? Atom %)
+                     (nil? (schema/check JRubyInstanceState @%)))
+               'JRubyInstanceState))
+
+;; A record representing an individual entry in the JRubyPuppet pool.
+(schema/defrecord JRubyPuppetInstance
+                  [pool :- pool-queue-type
+                   id :- schema/Int
+                   state :- JRubyInstanceStateContainer
+                   jruby-puppet :- JRubyPuppet
+                   scripting-container :- ScriptingContainer
+                   environment-registry :- (schema/both
+                                             EnvironmentRegistry
+                                             (schema/pred
+                                               #(satisfies? puppet-env/EnvironmentStateContainer %)))]
+                  Object
+                  (toString [this] (format "%s@%s {:id %s :state (Atom: %s)}"
+                                           "puppetlabs.services.jruby.jruby_puppet_core.JRubyPuppetInstance"
```

Comment:
Do you want to use the namespace in which this is defined here rather than "jruby_puppet_core"?